main.tf

```
@@ -67,8 +85,7 @@ resource "aws_ec2_tag" "cluster_primary_security_group" {
   # This should not affect the name of the cluster primary security group
   # Ref: https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2006
   # Ref: https://github.com/terraform-aws-modules/terraform-aws-eks/pull/2008
-  # `aws_default_tags` is merged in to "dedupe" tags and stabilize tag updates
-  for_each = { for k, v in merge(var.tags, var.cluster_tags, data.aws_default_tags.current.tags) :
+  for_each = { for k, v in merge(var.tags, var.cluster_tags) :
     k => v if local.create && k != "Name" && var.create_cluster_primary_security_group_tags
```

I think it is a good idea to add one condition to allow to delete tags if the value is `null`. Like this:

Like this:
```suggestion
    k => v if local.create && k != "Name" && var.create_cluster_primary_security_group_tags && v != null
```